util-core/src/main/scala/com/twitter/util/Promise.scala

```diff
@@ -467,26 +461,26 @@ class Promise[A] extends Future[A] with Promise.Responder[A] with Updatable[Try[
   // - Transforming
   // - Try[A] (Done)
   // - Promise[A]
-  @volatile private[this] var state: Any = WaitQueue.empty[A]
-  private def theState(): Any = state
+  private[this] val state: AtomicReference[Any] = new AtomicReference[Any](WaitQueue.empty[A])
```

Comment:
Hey @felixbr, was this an automated rewrite or did you change this yourself? Where you addressing a warning that looked like `the type test for <blah> cannot be checked at runtime`?

Reply:
I changed usage of `com.sun.misc.Unsafe` and `@volatile` to `AtomicReference` and other `Atomic*` types because the former no longer worked with Scala3 for me. I didn't try to fix or improve anything else because this code is very low-level and "dangerous". The tests all worked fine with `Atomic*` but I didn't do any performance checks.

There were some other changes in the file, simplifying the logic, which I've copied from earlier work by @martijnhoekstra. So maybe he was addressing that warning.

Reply:
Noted! thank you so much for expanding!